<launch>
  <arg name="role_name" default="ego_vehicle" />

  <arg name="path_length" default="200.0" />

  <arg name="town" default='Town03'/>

  <!-- depends on the scenario you want to test -->
  <!-- use comma separated format "x,y,z,roll,pitch,yaw" -->
<!-- used for HW3.2 -->
  <arg name="spawn_point" default='229.5,-86,2.5,0,0,-90'/>

  <!-- scenario :"MagicScenario_1" type:"SignalizedJunctionLeftTurn" town:Town03" -->
  <!-- <arg name="spawn_point" default='-123.7 , -137.4 , 2 , 0 , 0 , 0 '/>  -->

  <!-- scenario :"MagicScenario_2"  type:"DynamicObjectCrossing" town:Town03" -->
  <!-- <arg name="spawn_point" default='-73.8, -113 , 2, 0, 0, -90'/> -->

  <!-- scenario :"MagicScenario_3"  type:"OppositeVehicleRunningRedLight" town:Town03" -->
  <!-- <arg name="spawn_point" default='-73.9, -35, 2, 0, 0,-90'/> -->

  <!-- scenario :"MagicScenario_4" type:"VehicleTurningRight" town:Town03"> -->
  <!-- <arg name="spawn_point" default='-74.32, 80 , 2, 0, 0,-90'/> -->

  <!-- scenario :"MagicScenario_5" type:"ManeuverOppositeDirection" town:Town03"> -->
  <!-- <arg name="spawn_point" default='150, -193.2 , 3 , 0, 0,180'/> -->


  <!-- depends on the planner you want to use -->
  <!-- <arg name="planning_method" default='Optimal_Frenet_Planner'/>  -->
  
  <arg name='synchronous_mode_wait_for_vehicle_control_command' default='True'/> 
  <arg name='fixed_delta_seconds' default='0.02'/>



  <node pkg="planning" type="planning" name="planning" output="screen">
    <param name="role_name" type="string" value="$(arg role_name)" />
    <param name="path_length" type="double" value="$(arg path_length)" />

    <!-- depends on the planner you want to use -->
    <!-- <param name="planning_method" value="$(arg planning_method)" /> -->
    <!-- <rosparam command="load" file="$(find planning)/config/Optimal_Frenet_Planner.yaml"/>   -->

  </node>


  <node pkg="planning" type="cruise_speed_pub" name="cruise_speed_pub" output="screen"/>

  
  <include file="$(find carla_ros_bridge)/launch/carla_ros_bridge_with_example_ego_vehicle.launch">
    <arg name='spawn_point' value='$(arg spawn_point)'/>
    <arg name='town' value='$(arg town)'/>
    <arg name='synchronous_mode_wait_for_vehicle_control_command' value='$(arg synchronous_mode_wait_for_vehicle_control_command)'/>
    <arg name='fixed_delta_seconds' value='$(arg fixed_delta_seconds)'/>
  </include>

  
  <include file="$(find carla_waypoint_publisher)/launch/carla_waypoint_publisher.launch" />
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find planning)/rviz/planning.rviz"/>
</launch>